examples/src/bin/embassy_rmt_opentherm.rs

```diff
@@ -0,0 +1,338 @@
+            match i%2 {
+                0 => {
+                    data[i/2].level1 = entry;
+                    data[i/2].length1 = period;
+                },
+                _ => {
+                    data[i/2].level2 = entry;
+                    data[i/2].length2 = period;
+                }
+            }
+        }
+
+        for (i, entry) in data[..data.len()].iter().enumerate() {
+            println!("[{i}],e={},l={}, e2={},l2={}", entry.level1, entry.length1, entry.level2, entry.length2);
+        }
+
+        println!("transmit");
+        self.rmt_channel_tx.transmit(&data).await.unwrap();
+
+        //  let period = MANCHESTER_OPENTHERM_RESOLUTION;
+        //  self.output_pin.set_low(); //  generate idle state:
+        //  Timer::after(3 * convert_duration_to_embassy(period)).await; //  await one period in idle state
+
+        //  log::info!("Edge Trigger sent count: {count}");
+
+        //  self.output_pin.set_low(); //  generate idle state after
+        //  Timer::after(convert_duration_to_embassy(period)).await; //  await one period in idle state
+        Ok(())
+    }
+}
+
+// Inverted - invert polarity of the signal
+// N - output vec size
+impl<'pin, const N: usize, const Inverted: bool> EdgeCaptureInterface<N>
+    for RmtEdgeCapture<'pin, N, Inverted>
+{
+    //  TODO:
+    //  make a list
+    async fn start_capture(
+        &mut self,
+        timeout_inactive_capture: core::time::Duration,
+        timeout_till_active_capture: core::time::Duration,
+    ) -> Result<(InitLevel, Vec<core::time::Duration, N>), CaptureError> {
+
+        let init_state = match self.input_pin.is_high() {
+            true => InitLevel::High,
+            false => InitLevel::Low,
+        };
+
+        //  let mut capture_timestamp = start_timestamp;
+        //  let mut current_level = init_state.clone();
+        let mut timestamps = Vec::<core::time::Duration, N>::new();
+
+        const CAPTURE_DATA_SIZE: usize = VEC_SIZE_OT;
+        let mut capture_data = [PulseCode {
+            level1: true,
+            length1: 1,
+            level2: false,
+            length2: 1,
+        }; CAPTURE_DATA_SIZE];
+
+        //  let mut capture_data: [u32; CAPTURE_DATA_SIZE] = [0; CAPTURE_DATA_SIZE];
+
+        //  TODO: implement timeout or just handle it here:
+        let timestamps = match self.rmt_channel_rx.receive(&mut capture_data).await {
+            Ok(()) => {  //  Convert the data and return them
+                timestamps
+            },
+            Err(_) => { return Err(CaptureError::GenericError); }
+        };
+
+        //  todo!()  //  Assert
+
+        //  log::info!("Return InitLevel: {:?}", init_state);
+        Ok((init_state, timestamps))
+    }
+}
+
+const WIDTH: usize = 80;
+const RMT_CLK_DIV: u8 = 64;
+
+static RECEIVED_COUNT: Mutex<RefCell<Option<u32>>> =
+    Mutex::new(RefCell::new(None));
+
+static RECEIVED_DATA : critical_section::Mutex<RefCell<Option<[PulseCode; 48]>>> =
```

Comment:
If this isn't a different Mutex type than above, please do not specifiy the whole path to it.